pkg/spoke/controllers/manifestcontroller/manifestwork_controller.go

```diff
@@ -271,12 +293,57 @@ func (m *ManifestWorkController) applyOneManifest(
 	// TODO we should check the certain error.
 	// Use dynamic client when scheme cannot decode manifest or typed client cannot handle the object
 	if isDecodeError(result.Error) || isUnhandledError(result.Error) || isUnsupportedError(result.Error) {
-		result.Result, result.Changed, result.Error = m.applyUnstructured(ctx, unstructuredObj, gvr, recorder)
+		result.Result, result.Changed, result.Error = m.applyUnstructured(ctx, required, gvr, recorder)
 	}
 
 	return result
 }
 
+func (m *ManifestWorkController) serverSideApply(
+	ctx context.Context,
+	gvr schema.GroupVersionResource,
+	required *unstructured.Unstructured,
+	config *workapiv1.ServerSideApplyConfig,
+	recorder events.Recorder) (*unstructured.Unstructured, bool, error) {
+	force := false
+	fieldManager := "work-agent"
+
+	if config != nil {
+		force = config.Force
+		fieldManager = config.FieldManager
+	}
+
+	existing, err := m.spokeDynamicClient.
+		Resource(gvr).
+		Namespace(required.GetNamespace()).
+		Get(ctx, required.GetName(), metav1.GetOptions{})
+
+	switch {
+	case err == nil:
+		// merge ownerref by existing
```

Comment:
I didn't expect server-side apply to need this.

Reply:
I think this is needed when user changes the deleteOption to Orphan. We should remove the ownerref of ApplieManifestWork on this resource. Added a comment to explain this.